lib/r10k/git/errors.rb

```diff
@@ -10,24 +10,24 @@ class UnresolvableRefError < GitError
       attr_reader :ref
       attr_reader :git_dir
 
-      def initialize(*args)
+      def initialize(mesg, options = {})
+        if mesg.is_a? Hash
```

Comment:
Yeah, a lot of it was due to me trying to allow people to omit an explicit error message with https://github.com/puppetlabs/r10k/pull/270/files#diff-6645d47c52f4a2cd3f67ed2795e18f8bR23 but it's caused more heartache than it's worth.